File: backend/src/main/java/z9/hobby/domain/kakaoMap/dto/KakaoMapDto.java
package z9.hobby.domain.kakaoMap.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import z9.hobby.domain.classes.entity.ClassEntity;
import z9.hobby.model.schedules.SchedulesEntity;

public class KakaoMapDto {

    @Getter
    @AllArgsConstructor
    public static class SchedulesLocationData {
        private final Long id;
        private final String classTitle;
        private final String scheduleTitle;
        private final String favorite;
        private final String date;
        private final Double lat;
        private final Double lng;

        public static SchedulesLocationData from(
                ClassEntity classEntity,
                SchedulesEntity schedulesEntity
        ) {
            return new SchedulesLocationData(
                    classEntity.getId(),
                    classEntity.getName(),
                    schedulesEntity.getMeetingTitle(),
                    classEntity.getFavorite(),
                    schedulesEntity.getMeetingTime(),
                    schedulesEntity.getLat(),
                    schedulesEntity.getLng()
            );
        }
    }
}

File: backend/src/main/java/z9/hobby/domain/kakaoMap/dto/KakaoMapDto.kt
package z9.hobby.domain.kakaoMap.dto;

import z9.hobby.domain.classes.entity.ClassEntity
import z9.hobby.model.schedules.SchedulesEntity

class KakaoMapDto {

    data class SchedulesLocationData(
        val id: Long?,
        val classTitle: String,
        val scheduleTitle: String?,
        val favorite: String,
        val date: String?,
        val lat: Double?,
        val lng: Double?
    ) {
        companion object {
            @JvmStatic
            fun from(
                classEntity: ClassEntity,
                schedulesEntity: SchedulesEntity
            ): SchedulesLocationData {
                return SchedulesLocationData(
                    classEntity.id,
                    classEntity.name,
                    schedulesEntity.getMeetingTitle(),
                    classEntity.favorite,
                    schedulesEntity.getMeetingTime(),
                    schedulesEntity.getLat(),
                    schedulesEntity.getLng()
                );
            }
        }

    }
}